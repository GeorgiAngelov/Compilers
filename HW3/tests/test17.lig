var x : int = 4;

var b : bool;

function f(y : int) : bool {
      // Allowed.
      return (x == y);
      return (x != y);
      return (b == true);
      return (b != false);

      // Verboten.
      return (x == b);
      return (x != b);

      // Allowed.
      return ([1,2,3] == [1,2,3]);
      return ([1,2,3] != [1,2,3]);
      return ([1,2,3] == nil);
      return ([1,2,3] != nil);

      // Verboten.
      return ([1,2,3] > nil);
      return ([1,2,3] >= nil);
      return ([1,2,3] < nil);
      return ([1,2,3] <= nil);
      return (nil > nil);
      return (nil >= nil);
      return (nil < nil);
      return (nil <= nil);

      // Allowed.
      return ({a=nil} == {a=nil});
      return ({a=nil} != {a=nil});
      return ({a=nil} == nil);
      return ({a=nil} != nil);
      return ([1,2,3] == {a=nil});
      return ([1,2,3] != {a=nil});
      return (nil == nil);
      return (nil != nil);

      // Verboten.
      return ([1,2,3] == x);
      return ([1,2,3] != x);
      return ([1,2,3] == b);
      return ([1,2,3] != b);
}
