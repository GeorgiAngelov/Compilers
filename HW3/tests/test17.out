(decls
   (decl-var:ok (id x) (type-int) (num:int 4))
   (decl-var:ok (id b) (type-bool))
   (decl-fun:NONE (id f) (type-fun (params (typed-id (id y) (type-int))) (type-bool))
      (decls)
      (stmts
         (return:ok (==:bool (id:int x) (id:int y)))
         (return:ok (!=:bool (id:int x) (id:int y)))
         (return:ok (==:bool (id:bool b) (true:bool)))
         (return:ok (!=:bool (id:bool b) (false:bool)))
         (return:NONE (==:NONE (id:int x) (id:bool b)))
         (return:NONE (!=:NONE (id:int x) (id:bool b)))
         (return:ok (==:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3)))))
         (return:ok (!=:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3)))))
         (return:ok (==:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:ok (!=:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:NONE (>:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:NONE (>=:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:NONE (<:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:NONE (<=:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (nil:nil)))
         (return:NONE (>:NONE (nil:nil) (nil:nil)))
         (return:NONE (>=:NONE (nil:nil) (nil:nil)))
         (return:NONE (<:NONE (nil:nil) (nil:nil)))
         (return:NONE (<=:NONE (nil:nil) (nil:nil)))
         (return:ok (==:bool (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil)))) (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil))))))
         (return:ok (!=:bool (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil)))) (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil))))))
         (return:ok (==:bool (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil)))) (nil:nil)))
         (return:ok (!=:bool (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil)))) (nil:nil)))
         (return:ok (==:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil))))))
         (return:ok (!=:bool (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (struct-lit:{a : nil} (field-inits (field-init (id a) (nil:nil))))))
         (return:ok (==:bool (nil:nil) (nil:nil)))
         (return:ok (!=:bool (nil:nil) (nil:nil)))
         (return:NONE (==:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (id:int x)))
         (return:NONE (!=:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (id:int x)))
         (return:NONE (==:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (id:bool b)))
         (return:NONE (!=:NONE (array-lit:[int] (exps (num:int 1) (num:int 2) (num:int 3))) (id:bool b))))))
Function main() not defined or has wrong type!
Typechecking failed!
